tools/query_verification/src/main/java/com/google/bigquery/QueryVerifier.java

```diff
@@ -38,7 +44,48 @@ public void verify() {
     public void verifyDataFree() {
         boolean verificationResult = false;
 
-        // TODO Implement data free verification
+        BigQuery bigQuery = BigQueryOptions.getDefaultInstance().getService();
+
+        List<Table> tables = new ArrayList<Table>();
+
+        // Create tables based on schema
+        if (migratedSchema != null) {
+            try {
+                if (migratedSchema.isInJsonFormat()) {
+                    TableInfo tableInfo = QueryVerifier.getTableInfoFromJsonSchema(migratedSchema);
+                    Table table = bigQuery.create(tableInfo);
+                    tables.add(table);
+                } else {
+                    // TODO Load schema from DDL
+                }
+            } catch (NullPointerException e) {
+                System.out.println(migratedSchema.path() + " is not correctly formatted.");
+            }
+        }
+
+        // Create dry-run job
+        QueryJobConfiguration queryConfig = QueryJobConfiguration.newBuilder(migratedQuery.query())
+                .setDryRun(true)
+                .build();
+        JobId jobId = JobId.of(UUID.randomUUID().toString());
+        JobInfo jobInfo = JobInfo.newBuilder(queryConfig)
+                .setJobId(jobId)
+                .build();
+
+        // TODO Support multiple queries
+
+        try {
+            // Run dry-run
+            Job queryJob = bigQuery.create(jobInfo);
+            verificationResult = queryJob.getStatistics() != null;
```

Comment:
I assume in the case of a bad query we'd get back some error information from queryJob, it's probably worth collecting that if we can.